pueue_lib/src/network/socket/unix.rs

```diff
@@ -142,8 +142,19 @@ pub async fn get_listener(settings: &Shared) -> Result<GenericListener, Error> {
             })?;
         }
 
+        // Adjust the umask prior to creating the socket.
+        let old_umask = settings
+            .unix_socket_permissions
+            .map(|permissions| unsafe { libc::umask(!permissions) });
+
         let unix_listener = UnixListener::bind(&socket_path)
-            .map_err(|err| Error::IoPathError(socket_path, "creating unix socket", err))?;
+            .map_err(|err| Error::IoPathError(socket_path.clone(), "creating unix socket", err))?;
+
+        // Reset the umask now that the socket is created.
+        if let Some(permissions) = old_umask {
+            unsafe { libc::umask(permissions) };
```

Comment:
Another problem with this approach is that Pueue is multi-threaded.

From what I understand, setting the `umask` affects the whole process.
And the umask is also inherited to child processes.

-> We have to make really sure that 
- No files are written while this happens
- No subprocesses of any kind are started while this happens.